src/components/ChartView/ChartDeployButton.tsx

```diff
@@ -25,8 +37,18 @@ class ChartDeployButton extends React.Component<IChartDeployButtonProps, IChartD
     modalIsOpen: false,
     namespace: "default",
     releaseName: "",
+    values: "",
+    valuesModified: false,
   };
 
+  public componentWillReceiveProps(nextProps: IChartDeployButtonProps) {
```

Comment:
out of curiosity, when would the props change?

Reply:
At initial mounting of `ChartDeployButton`, takes the `values` prop it is provided and applies it as `values` in state. During the initial mounting `ChartViews` `values` might not have finished fetching yet so `ChartDeployButton` will receive an empty string. `ChartDeployButton` will receive a new set of props when the `ChartView` finishes fetching the values.yml (or any time `ChartView` receives new props really)

Reply:
Oh right, yes that makes sense. Thanks!